src/components/titlePane/createTitlePane.ts

```diff
@@ -0,0 +1,88 @@
+import { VNodeProperties } from '@dojo/interfaces/vdom';
+import createWidgetBase from '@dojo/widget-core/createWidgetBase';
+import { v } from '@dojo/widget-core/d';
+import { DNode, Widget, WidgetFactory, WidgetProperties } from '@dojo/widget-core/interfaces';
+import themeable, { ThemeableMixin } from '@dojo/widget-core/mixins/themeable';
+
+import * as css from './styles/titlePane.css';
+import * as animations from '../../styles/animations.css';
+
+export interface TitlePaneProperties extends WidgetProperties {
+	collapsed?: boolean;
+	collapsible?: boolean;
+	enterAnimation?: string;
+	exitAnimation?: string;
+	title: string;
+
+	onRequestCollapse?(): void;
+	onRequestExpand?(): void;
+};
+
+export type TitlePane = Widget<TitlePaneProperties> & ThemeableMixin & {
+	onClickTitle?(): void;
+};
+
+export interface TitlePaneFactory extends WidgetFactory<TitlePane, TitlePaneProperties> { };
+
+const createTitlePane: TitlePaneFactory = createWidgetBase
+.mixin(themeable)
+.mixin({
+	mixin: {
+		baseClasses: css,
+
+		onClickTitle: function (this: TitlePane) {
+			const {
+				properties: {
+					collapsed = false
+				}
+			} = this;
+
+			if (collapsed) {
+				this.properties.onRequestExpand && this.properties.onRequestExpand();
+			}
+			else {
+				this.properties.onRequestCollapse && this.properties.onRequestCollapse();
+			}
+		},
+
+		getChildrenNodes: function (this: TitlePane): DNode[] {
```

Comment:
If you rebase against master, widgets will be using the most recent release of widget-core. Now, `getChildrenNodes` is replaced by a `render` function. This function should return a single top-level `DNode`, the children of which should be what you're currently returning here. See the Dialog or SlidePanel for reference.